### PR TITLE
Adds validation of Parameters

### PR DESCRIPTION
**Pull request by alanbo**
*Issue #, if available:*
#26 

*Description of changes:*
I added logic checking the properties of Parameters that end with `.$` to be JSON Path. At the moment, I'm only checking that the value is a string beginning with `$`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/constants/diagnosticStrings.ts [hunk 1]
```diff
@@ -7,8 +7,9 @@ export const MESSAGES = {
     INVALID_NEXT: 'The value of "Next" property must be the name of an existing state.',
     INVALID_DEFAULT: 'The value of "Default" property must be the name of an existing state.',
     INVALID_START_AT: 'The value of "StartAt" property must be the name of an existing state.',
+    INVALID_JSON_PATH: 'The value for the field must be a valid JSONPath',
```
Comment (jormello) [inline src/constants/diagnosticStrings.ts:10]: All other error messages end with a period.

File: src/validation/validateStates.ts [hunk 2]
```diff
@@ -47,6 +47,30 @@ interface ValidateCatchResult {
...
+function validateParameters(parametersPropNode: PropertyASTNode, document: TextDocument): Diagnostic[] {
+    let diagnostics: Diagnostic[] = []
+    const valueNode = parametersPropNode.valueNode
+
+    if (isObjectNode(valueNode)) {
+        valueNode.properties.forEach(prop => {
+            if (prop.keyNode.value.endsWith('.$')) {
+                const propValue = prop.valueNode.value
+
+                if (typeof propValue !== 'string' || !propValue.startsWith('$')) {
```
Comment (jormello) [inline src/validation/validateStates.ts:59]: Do we want to trim? Not sure whether the JSON path is still valid if prefixed with spaces.

Comment (alanbo) [reply src/validation/validateStates.ts:59]: That's a very valid question. I checked in StepFunctions console. We don't need to trim - it always starts with $.

File: src/constants/diagnosticStrings.ts (hunk 1 above)
Comment (alanbo) [reply src/constants/diagnosticStrings.ts]: ok, done.

**Pull request merged**